projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -135,15 +144,21 @@ export class TuiInputPhoneInternationalComponent
         this.close();
     }
 
+    onPaste(event: ClipboardEvent) {
+        const value = getClipboardDataText(event);
+        const country = this.countries.find(countryIsoCode =>
+            value.startsWith(this.isoToCountryCode(countryIsoCode)),
+        );
+
+        if (country) {
+            this.countryIsoCode = country;
+            this.updateValue('+' + value.replace(TUI_NON_DIGITS_REGEXP, ''));
+        }
+    }
+
     @tuiPure
     isoToCountryCode(isoCode: TuiCountryIsoCode): string {
-        return (
-            '+' +
-            COUNTRIES_MASKS[isoCode]
-                .split('')
-                .filter(symbol => TUI_DIGIT_REGEXP.test(symbol))
-                .join('')
-        );
+        return COUNTRIES_MASKS[isoCode].replace(MASK_AFTER_CODE_REGEXP, '');
```

Comment:
I wonder how performant this operation is. Maybe it's better to use tuiMapper pipe

Reply:
thanks, cpu usage seems to be better now